docs/services/ml/upload-training-script.md

```diff
@@ -0,0 +1,216 @@
+Parse these arguments in your training script as follows:
+
+```python {class="line-numbers linkable-line-numbers" data-line="364"}
+# This parses the required args for running the training script.
+# You can use the data JSON to parse the images and
+# their annotations for the specified dataset.
+# You can use the model directory for saving model artifacts
+# associated with the model name and version.
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--dataset_file", dest="data_json", type=str)
+    parser.add_argument("--model_output_directory", dest="model_dir", type=str)
+    args = parser.parse_args()
+    model_dir = args.model_dir
+    data_json = args.data_json
+    return model_dir, data_json
+```
+
+{{% /expand%}}
+{{%expand "Step 2: Parse annotations from dataset file" %}}
+
+The `dataset_file` is a file generated by Viam containing the file path and annotations associated with the dataset ID on which the user is training.
+The annotations are formatted as follows:
+
+```json {class="line-numbers linkable-line-numbers"}
+{"image_path": "/path/to/data/data1.jpeg", "bounding_box_annotations":[{"annotation_label":"blue_star","x_min_normalized":0.38175675675675674,"x_max_normalized":0.5101351351351351,"y_min_normalized":0.35585585585585583,"y_max_normalized":0.527027027027027}], "classification_annotations": [{"annotation_label": "blue_star"}]}
+{"image_path": "/path/to/data/data2.jpeg", "bounding_box_annotations":[{"annotation_label":"blue_star","x_min_normalized":0.2939189189189189,"x_max_normalized":0.4594594594594595,"y_min_normalized":0.25225225225225223,"y_max_normalized":0.5495495495495496}], "classification_annotations": [{"annotation_label": "blue_star"}]}
+{"image_path": "/path/to/data/data3.jpeg", "bounding_box_annotations":[{"annotation_label":"blue_star","x_min_normalized":0.03557312252964427,"x_max_normalized":0.2015810276679842,"y_min_normalized":0.w30526315789473685,"y_max_normalized":0.5368421052631579}, {"annotation_label":"blue_square","x_min_normalized":0.039525691699604744,"x_max_normalized":0.2015810276679842,"y_min_normalized":0.2578947368421053,"y_max_normalized":0.5473684210526316}], "classification_annotations": [{"annotation_label": "blue_star"}, {"annotation_label": "blue_square"}]}
+```
+
+With classification annotations like this:
+
+```json {class="line-numbers linkable-line-numbers"}
+"classification_annotations": [{"annotation_label": "blue_star"}, {"annotation_label": "blue_square"}]}
+```
+
+And bounding box annotations looking like this:
+
+```json {class="line-numbers linkable-line-numbers"}
+"bounding_box_annotations":[{"annotation_label":"blue_star","x_min_normalized":0.38175675675675674,"x_max_normalized":0.5101351351351351,"y_min_normalized":0.35585585585585583,"y_max_normalized":0.527027027027027}]
+```
```

Comment:
I don't think this is actually needed. If we want to make the above more legible by formatting it better that would be better
```suggestion
```

Reply:
That is a valid example. The full line will have some mix of classification annotation and bounding box annotations, but each annotation is formatted differently and likely only one of them would be used per training script